experimental/dds/tree/src/id-compressor/IdCompressor.ts

```diff
@@ -1189,15 +1180,14 @@ export class IdCompressor {
 				const inversionKey = IdCompressor.createInversionKey(override);
 				const compressionMapping =
 					this.clustersAndOverridesInversion.get(inversionKey) ?? fail('Bimap is malformed.');
-				if (
-					!IdCompressor.isClusterInfo(compressionMapping) &&
+				return !IdCompressor.isClusterInfo(compressionMapping) &&
 					!IdCompressor.isUnfinalizedOverride(compressionMapping) &&
 					compressionMapping.associatedLocalId === id
-				) {
-					return compressionMapping.originalOverridingFinal;
-				}
+					? compressionMapping.originalOverridingFinal
+					: (id as OpSpaceCompressedId);
 			}
-			return id as OpSpaceCompressedId;
+			const possibleFinal = this.sessionIdNormalizer.getFinalId(id);
+			return possibleFinal === undefined ? (id as OpSpaceCompressedId) : possibleFinal[0];
```

Comment:
```suggestion
			return possibleFinal?.[0] ?? id as OpSpaceCompressedId;
```
You don't need to make this change, but FYI `?.[]` is a thing.

Reply:
Done!